AAE-37399 Add root process instance id to task

mcoune opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 The commit message follows our guidelines
     - 👷‍♂️ Tests for the changes have been added (for bug fixes / features)
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.
-->

<!--
(check one with "x") one of the following
-->

**What kind of change does this PR introduce?**

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

## Description

Extend task model with root process instance id to use it in analytics and openSearch

- Issue Link:
[JIRA TICKET](https://hyland.atlassian.net/browse/AAE-37399?atlOrigin=eyJpIjoiMmI4NmM1YWFkMjYxNDFmMThhZDFhNmVjNzgxNWRhZTQiLCJwIjoiaiJ9)

<!--
If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
-->

**Does this PR introduce a breaking change?** (check one with "x")

> - [X] Yes
> - [ ] No

mcoune committed:
Adds root process instance id to Task

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Activiti/Activiti?pullRequest=5111) <br/>All committers have signed the CLA.

mcoune commented on activiti-api/activiti-api-task-model/src/main/java/org/activiti/api/task/model/Task.java:
```diff
@@ -15,20 +15,18 @@
         COMPLETED,
         CANCELLED,
-        DELETED
+        DELETED,
```
used prettier.

mcoune committed:
Merge branch 'develop' into feature/AAE-37399-add-root-process-instance-id-to-task

fkindgen committed:
Speed up build by skipping tests
Renames root process instance id property

Renames process root instance id field

Renames task root process instance id property

Update activiti-core/activiti-engine/src/main/java/org/activiti/engine/task/Task.java

Co-authored-by: Michaela <85624192+mkrbr@users.noreply.github.com>
Adds root process instance id to task entity

reverted changes

removed unused import

prettier

fkindgen commented on .github/workflows/main_pull.yml:
```diff
@@ -89,7 +89,7 @@ jobs:
         run: mvn ${{ steps.define_maven_command.outputs.command }} ${{ env.MAVEN_CLI_OPTS}}
         env:
-          MAVEN_CLI_OPTS: --show-version --no-transfer-progress --settings settings.xml
+          MAVEN_CLI_OPTS: -DskipTests -DskipITs -T 8C --show-version --no-transfer-progress --settings settings.xml
```
Please revert before merge

mkrbr commented on activiti-core/activiti-engine/src/main/java/org/activiti/engine/task/Task.java:
```diff
@@ -80,6 +80,9 @@ public interface Task extends TaskInfo {
   /** Change the form key of the task */
   void setFormKey(String formKey);
 
+    /** Id ot the root process */
```
```suggestion
    /** ID of the root process */
```

mcoune commented on activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/TaskEntityImpl.java:
Prettier formated the whole class, watch out for changes related with 'taskProcessRootProcessInstanceId'

mcoune commented:
Prettier formatted a lot of lines, watch out for changes related with 'taskProcessRootProcessInstanceId'

mcoune committed:
prettier

mcoune committed:
rem prettier

mcoune committed:
reverted change

mcoune committed:
Merge branch 'develop' into feature/AAE-37399-add-root-process-instance-id-to-task

mcoune committed:
removed unused import

mcoune committed:
Merge remote-tracking branch 'origin/feature/AAE-37399-add-root-process-instance-id-to-task' into feature/AAE-37399-add-root-process-instance-id-to-task
